dao/src/main/java/org/thingsboard/server/dao/rule/BaseRuleChainService.java

```diff
@@ -758,6 +797,44 @@ public void deleteRuleNodes(TenantId tenantId, RuleChainId ruleChainId) {
         deleteEntityRelations(tenantId, ruleChainId);
     }
 
+    @Override
+    public ListenableFuture<RuleNodeStats> getRuleNodeStats(TenantId tenantId, RuleNodeId ruleNodeId) {
+        return Futures.transform(attributesService.find(tenantId, ruleNodeId, DataConstants.SERVER_SCOPE, "ruleNodeStats"), input ->
+                input.map(attributeKvEntry -> JacksonUtil.fromString(attributeKvEntry.getJsonValue().get(), RuleNodeStats.class))
+                        .orElseGet(RuleNodeStats::new), MoreExecutors.directExecutor());
```

Comment:
**MoreExecutors.directExecutor()**

In this case, thread from the actor system is going to be used to save and get attributes from the database. 
This is totally wrong as affect performance of the actor system. 
Please introduce or re-use some other executor - dbCallBackExecutor service or similar